index.js

```diff
@@ -13,21 +15,34 @@ function ZipCodeApi() {
 
 ZipCodeApi.prototype.init = function(options) {
   options = options || {};
+  const clientKey = options.clientKey;
   const apiKey = options.apiKey;
-
-  if (!apiKey) {
-    throw new Error('Must set an API key');
+  const domain = options.domain;
+  if (!clientKey || !apiKey || !domain) {
+    throw new Error('Must set a client key, api key, and domain');
   }
 
+  this.clientKey = clientKey;
   this.apiKey = apiKey;
+  this.domain = domain;
 };
 
 ZipCodeApi.prototype.makeRestUrl = function(suffix) {
-  if (!this.apiKey) {
-    throw new Error('Must set an API key');
+  if (!this.clientKey || !this.apiKey || !this.domain) {
+    throw new Error('Must set a client key, api key, and domain');
   }
 
-  return 'https://www.zipcodeapi.com/rest/' + this.apiKey + suffix;
+  const salt = 'Abcd1234'
+  const currentTime = Math.floor(Date.now() / 1000);
+  const sigPrefix = `${currentTime}-6-${salt}`;
+  const sigBase = `${sigPrefix}-${this.domain.toLowerCase()}-${this.apiKey}`;
+  const hash = sha1(sigBase);
+  const authHash = `${sigPrefix}-${hash}`;
```

Comment:
I've tried this using sigBase and sigPrefix. The documentation says to use sigBase but the example they have is using sigPrefix.

https://www.zipcodeapi.com/ClientKeySetup/8804